hardware/nao/nao_service.py

```
@@ -0,0 +1,229 @@
+        
+        # Send response back to GUI
+        response = "SUCCESS" if success else "FAILED"
+        print("Sending response:", response)
+        
+        # Ensure response is sent properly
+        if isinstance(response, str):
+            response = response.encode('utf-8')
+        
+        client.sendall(response)
+        print("Response sent successfully")
+        return True
```

Ruff (0.14.2)

Consider moving this statement to an `else` block

(TRY300)